connectors/kafka/src/main/java/forklift/message/MessageStream.java

```diff
@@ -0,0 +1,70 @@
+package forklift.message;
+
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A stream of available messages which can be retrieved by topic.
+ */
+public class MessageStream implements ReadableMessageStream {
+    private static final Logger log = LoggerFactory.getLogger(MessageStream.class);
+    Map<String, BlockingQueue<KafkaMessage>> topicQueue = new ConcurrentHashMap<>();
+
+    /**
+     * Adds the passed in records to the stream.  After being added, a record is available to be retreived through a
+     * call to {@link #nextRecord(String, long)}.
+     * <p>
+     * <strong>Note:</strong> All passed in records must belong to a topic added through a call to {@link #addTopic(String)}.
+     *
+     * @param records the records to add
+     * @throws IllegalStateException if the capacity of a stream has been exceeded.
+     */
+    public void addRecords(Map<TopicPartition, List<KafkaMessage>> records) {
+        log.debug("Adding records to stream");
+        for (Map.Entry<TopicPartition, List<KafkaMessage>> entry : records.entrySet()) {
+            for (KafkaMessage message : entry.getValue()) {
+                BlockingQueue<KafkaMessage> queue = topicQueue.get(entry.getKey().topic());
```

Comment:
Minor optimization: grabbing the blocking queue for a given topic can be done in the higher loop and then the `queue != null` check can be done earlier as well.